chainforge/react-server/src/backend/backend.ts

```diff
@@ -711,11 +726,18 @@ export async function queryLLM(id: string,
     // Await the responses from all queried LLMs
     const llm_results = await Promise.all(tasks);
     llm_results.forEach(result => {
+      if (QueryTracker.has(id)) {
+        console.log("HIT 3")
+        throw new Error("CANCELLED");
+      }
```

Comment:
This is also not a good time to cancel, because we only reach this state *after* all responses have returned! In other words, there's no point in canceling, because we're done! 

Can you revert lines 729-808 back to what they were? I.e. the entire rest of the function.